code/modules/mob/living/simple_animal/hostile/retaliate/nymph.dm

```diff
@@ -175,18 +175,22 @@
 	if(isdiona(arrived))
 		if(mind != null || stat == DEAD || is_drone) //Does the nymph on the ground have a mind, dead or a drone?
 			return // If so, ignore the diona
-		var/mob/living/carbon/human/H = arrived
-		var/list/limbs_to_heal = H.get_missing_limbs()
+		var/mob/living/carbon/human/arrived_diona = arrived
+		var/list/limbs_to_heal = arrived_diona.get_missing_limbs()
 		if(!LAZYLEN(limbs_to_heal))
 			return
-		playsound(H, 'sound/creatures/venus_trap_hit.ogg', 25, 1)
+		toggle_ai(AI_OFF)
+		if(!do_after(arrived_diona, 5 SECONDS, source, progress = TRUE))
```

Comment:
This proc doesn't have the SIGNAL_HANDLER define in error. If you add it you will find that this line fails because signal handlers shouldn't be sleeping as they cause unpredictable behaviour. You will need to mark this proc as a signal handler and either call this behaviour asynchronously or set the proc to waitfor false.